server/schedule/checker/replica_strategy.go

```diff
@@ -53,6 +54,7 @@ func (s *ReplicaStrategy) SelectStoreToAdd(coLocationStores []*core.StoreInfo, e
 
 	filters := []filter.Filter{
 		filter.NewExcludedFilter(s.checkerName, nil, s.region.GetStoreIds()),
+		filter.NewIsolationFilter(s.checkerName, s.isolationLevel, s.locationLabels, s.cluster.GetRegionStores(s.region)),
```

Comment:
should use `coLocationStores` instead of `s.cluster.GetRegonStores(s.region))` here.